/*
 *  Copyright (c) 2016, https://github.com/nebula-im
 *  All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package zproto;

option java_package = "com.zchat.engine.zproto";
option java_outer_classname = "ZProto";
option optimize_for = CODE_SIZE;

// 获取下一个Seq
// GetNextSequenceReq -> SequenceRsp
message FetchNextSequenceReq {
  uint32 id = 1;
}

// 查询当前Seq
// GetCurrentSequenceReq -> SequenceRsp
message GetCurrentSequenceReq {
  uint32 id = 1;
}

// 返回
message SequenceRsp {
  uint64 sequence = 1;
}

// FetchNextSequenceListReq -> SequenceListRsp
message FetchNextSequenceListReq {
  repeated uint32 id_list = 1;
}

// 查询当前Seq
// GetCurrentSequenceListReq -> SequenceListRsp
message GetCurrentSequenceListReq {
  repeated uint32 id_list = 1;
}

message IdSeq {
  uint32 id = 1;
  uint64 sequence = 2;
}

// SequenceListRsp
message SequenceListRsp {
  repeated IdSeq sequence_list = 1;
}

// 加载max_seqs数据
// LoadMaxSeqsDataReq -> LoadMaxSeqsDataRsp
message LoadMaxSeqsDataReq {
  uint32 set_id = 1;
  uint32 alloc_id = 2;
}

// LoadMaxSeqsDataRsp
message LoadMaxSeqsDataRsp {
  bytes max_seqs = 1;
}

// 加载max_seqs数据
// SaveMaxSeqReq -> SaveMaxSeqRsp
message SaveMaxSeqReq {
  uint32 set_id = 1;
  uint32 alloc_id = 2;
  uint32 section_id = 3;
  uint64 max_seq = 4;
}

// SaveMaxSeqRsp
message SaveMaxSeqRsp {
  uint64 last_max_seq = 1;
}
